### PR TITLE
[https://jira.qos.ch/browse/LOGBACK-1595] add JANSI 2 support

### PR DESCRIPTION
**Pull request by bmarwell**
This PR will make logback support both JANSI1 and JANSI2.

* [LOGBACK-1567](https://jira.qos.ch/browse/LOGBACK-1567) -- Upgrade to jansi 2.3.4
* [LOGBACK-1595](https://jira.qos.ch/browse/LOGBACK-1595) -- Update Jansi library

Also see supersedes https://github.com/qos-ch/logback/pull/526 / supersedes https://github.com/qos-ch/logback/pull/527 / supersedes https://github.com/qos-ch/logback/pull/542

### CODE REVIEW

File: logback-core/src/main/java/ch/qos/logback/core/ConsoleAppender.java [hunk 1]
```diff
@@ -94,6 +96,20 @@ private OutputStream wrapWithJansi(OutputStream targetStream) {
             addInfo("Enabling JANSI AnsiPrintStream for the console.");
             ClassLoader classLoader = Loader.getClassLoaderOfObject(context);
             Class<?> classObj = classLoader.loadClass(AnsiConsole_CLASS_NAME);
+
+            // check for JAnsi 2
+            final Optional<Method> optOutMethod = Arrays.stream(classObj.getMethods())
+                    .filter(m -> m.getName().equals("out"))
```
Comment (gnodet) [inline logback-core/src/main/java/ch/qos/logback/core/ConsoleAppender.java:102]: I think this should check wether the `target == ConsoleTarget.SystemOut` to look for either `out` or `err` method.

Comment (bmarwell) [reply logback-core/src/main/java/ch/qos/logback/core/ConsoleAppender.java:102]: You are right! Will fix this.

**Review by gnodet — APPROVED** [review]
Looks good to me.  It would be nice to have that long awaited Jansi 2.x support.

**Comment by gnodet** [comment]
@ceki

**Comment by ceki** [comment]
@gnodet  I has not seen that [PR 542](https://github.com/qos-ch/logback/pull/542) provides a solution of the System.out interception.

**Pull request merged**